test/framework/controlplane.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	corev1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// MultiNodeControlplaneClusterInput defines the necessary dependencies to run a multi-node control plane cluster.
+type MultiNodeControlplaneClusterInput struct {
+	Mgmt              ManagementCluster
+	Cluster           *corev1.Cluster
+	InfraCluster      runtime.Object
+	ControlplaneNodes []Node
+	CreateTimeout     time.Duration
+}
+
+// SetDefaults defaults the struct fields if necessary.
+func (m *MultiNodeControlplaneClusterInput) SetDefaults() {
+	if m.CreateTimeout == 0*time.Second {
```

Comment:
```suggestion
	if m.CreateTimeout == 0 {
```

does this work?

Reply:
https://play.golang.org/p/UGxXPBfEgw6 it sure does!